### DISCUSSION TITLE
How to run the sample projects?

### DISCUSSION THREAD

**Discussion by tdkehoe** [angular/angularfire#2969] [Q&A]
I want to take a look at the sample projects. I cloned the angularfire repo, went into the folder `angularfire/samples/compat`, and ran `ng serve`. I got this error message:

```An unhandled exception occurred: Cannot find module '@angular-devkit/build-angular/package.json'```

I installed the module:

```npm install --save-dev @angular-devkit/build-angular```

but got another error message 

```npm ERR! ERESOLVE unable to resolve dependency tree```

Something about jasmine-core.

What am I doing wrong? Do the sample projects run?

**Comment by jamesdaniels** [comment]
the samples are currently using yarn. just cd into them and run yarn, yarn start

**Reply by tdkehoe** [reply]
The [README](https://github.com/angular/angularfire/blob/master/samples/compat/README.md) file doesn't say anything about yarn.

**Reply by jamesdaniels** [reply]
That README is the placeholder generated by the Angular CLI on a fresh project.

**Comment by tdkehoe** [comment]
I installed yarn, upgraded to the latest version, ran `yarn install`, and then ran `yarn start`. I get an error message 

`Usage Error: Couldn't find a script named "start".`

I've never used yarn. 'yarn help` doesn't show that `yarn start` is a command for yarn. I tried 'yarn run` and it listed a bunch of green messages, which seems good, but no port opened for the project. What should I do now?

**Reply by jamesdaniels** [reply]
Yarn is the same as NPM, expect you can skip the run, e.g, `yarn start` is the same as `npm run start`. Works fine for me 🤷 even on a fresh checkout.

In the root, `yarn install` then `yarn build`, this builds AngularFire.
Then cd into the sample that you want to use `cd samples/compat`, run `yarn install` and then `yarn start`
